core/sbt-cloudflow/src/main/scala/cloudflow/sbt/CloudflowLocalRunnerPlugin.scala

```diff
@@ -199,25 +196,67 @@ object CloudflowLocalRunnerPlugin extends AutoPlugin {
     s"$org/$name"
   }
 
-  case class RuntimeDescriptor(id: String,
-                               appDescriptor: ApplicationDescriptor,
-                               appDescriptorFile: Path,
-                               outputFile: File,
-                               logConfig: Path,
-                               localConfPath: Option[String])
+  val kafka = new AtomicReference[KafkaContainer]()
 
-  def setupKafka(port: Int, topics: Seq[String])(implicit log: Logger) = {
-    implicit val kafkaConfig = EmbeddedKafkaConfig(kafkaPort = port)
-    EmbeddedKafka.start()
-    log.debug(s"Setting up embedded Kafka broker on port: $port")
+  def setupKafka(topics: Seq[String])(implicit log: Logger) = {
+
+    val cl = Thread.currentThread().getContextClassLoader()
+    val kafkaPort =
+      try {
+        val c = getClass().getClassLoader()
+        Thread.currentThread().setContextClassLoader(c)
+
+        val k = new KafkaContainer(tcutility.DockerImageName.parse("confluentinc/cp-kafka:5.4.3"))
+          .withExposedPorts(KafkaPort)
+        k.start()
+        kafka.set(k)
+
+        k.getMappedPort(KafkaPort)
+      } finally {
+        Thread.currentThread().setContextClassLoader(cl)
+      }
+
+    log.debug(s"Setting up embedded Kafka broker on port: $kafkaPort")
 
     topics.foreach { topic =>
       log.debug(s"Kafka Setup: creating topic: $topic")
-      EmbeddedKafka.createCustomTopic(topic)
+
+      import org.apache.kafka.clients.admin.{ AdminClient, AdminClientConfig, NewTopic }
+      import scala.collection.JavaConverters._
+
+      val newTopic = new NewTopic(topic, 1, 1.toShort)
+
+      val adminClient = AdminClient.create(
+        Map[String, Object](
+          AdminClientConfig.BOOTSTRAP_SERVERS_CONFIG -> s"localhost.:${kafkaPort}",
+          AdminClientConfig.CLIENT_ID_CONFIG         -> "embedded-kafka-admin-client"
+        ).asJava
+      )
```

Comment:
You can create the admin client outside of the loop.

Reply:
correct